### ISSUE TITLE
[java] UnnecessaryLocalBeforeReturn false positive when incrementing counter

### ISSUE THREAD

**Issue by mkordas** [pmd/pmd#335]
**Rule Set: design**

**Description:**

`UnnecessaryLocalBeforeReturn` doesn't allow to increment variable before returning from method

**Code Sample demonstrating the issue:**

```java
        String newId = prefix + number;
        ++number;
        return newId;
```

Violation:
```
[UnnecessaryLocalBeforeReturn] Consider simply returning the value vs storing it in local variable 'newId'

**Comment by jsotuyod** [comment]
@mkordas thanks for the report.

This is a duplicate of #282, which was fixed on #294. Will be part of the PMD 5.6.0 release.
